### PR TITLE
[WIP] feat: add support pure junit3 projects

### PR DESCRIPTION
**Pull request by danglotb**

### CODE REVIEW

**Comment by danglotb** [comment]
Hi

Discovered that test-runner requires JUnit4 to run, even if the tests are in JUnit3.

Instead of duplicating all the runners for JUnit3 (such as the ones for JUnit5), I propose to inject JUnit4 (and hamcrest) in the provided classpath.

WDYT @andre15silva @martinezmatias @monperrus? I need to make the same changes others runs (coverage, online, etc.)

File: src/main/java/eu/stamp_project/testrunner/EntryPoint.java [hunk 1]
```diff
@@ -257,6 +259,19 @@ public static TestResult runTests(String classpath, String[] fullQualifiedNameOf
         return EntryPoint.runTests(javaCommand);
     }
 
+    private static String checkAndAddJUnit4(String classpath) {
+        if (!classpath.contains("junit-4")) {
```
Comment (andre15silva) [inline src/main/java/eu/stamp_project/testrunner/EntryPoint.java:263]: I'm not sure this is the best way of checking whether Junit4 is in the dependencies, because it can be included in a subdirectory or jar file.

An alternative could be to launch a new thread with a custom classloader (initialized with the original classpath), and then checking in runtime whether the relevant classes are loaded.

WDYT @danglotb ?

**Comment by andre15silva** [comment]
Other than that comment, I agree with the changes!

File: src/main/java/eu/stamp_project/testrunner/EntryPoint.java (hunk 1 above)
Comment (danglotb) [reply src/main/java/eu/stamp_project/testrunner/EntryPoint.java:263]: Thank for the feebacks. However, it seems very costly to launch a new Thread (and I guess a JVM, since we want to check if the required classes are properly loaded).

I mean, in 99.9% of cases, you will launch a new Thread to perform this check for "nothing" (JUnit4 will be loaded). I think that this is a corner case and it should not slow down the main use cases of test-runner.

I have no evidence of my claim, it is what my guts tell me :smile:

Comment (andre15silva) [reply src/main/java/eu/stamp_project/testrunner/EntryPoint.java:263]: I understand, but doing so in this fashion might introduce dependency conflicts (e.g. different JUnit4 versions) no?

Comment (danglotb) [reply src/main/java/eu/stamp_project/testrunner/EntryPoint.java:263]: I do not think so since the original classpath does not have JUnit4, right? So injecting our own version of JUnit4 cannot result in dependency conflict. 

Or I might not understand the underlying problem here :sweat_smile:

**Comment by danglotb** [comment]
It seems that there is the same issue with JUnit5.

It is required to have `org.junit.platform:junit-platform-launcher`, in particular the class `org.junit.platform.launcher.TestExecutionListener`.

File: src/main/java/eu/stamp_project/testrunner/EntryPoint.java (hunk 1 above)
Comment (andre15silva) [reply src/main/java/eu/stamp_project/testrunner/EntryPoint.java:263]: Well, you check whether the classpath contains the sub-string `junit-4` or not.

But there could be the case where you have a JUnit4 in the classpath but under a different name (e.g. inside a .jar, or in a subdirectory).

**Comment by danglotb** [comment]
This PR is not valid anymore, please see #144.
